connectors/source.py

```diff
@@ -414,6 +413,11 @@ def hash_id(cls, _id):
 
         return hash_id(_id)
 
+    @classmethod
+    def features(cls):
+        """Returns features available for the data source"""
+        raise NotImplementedError
+
```

Comment:
👍 
My original instinct was that this should have an implementation, to limit how many concrete data sources needed to override this function. The more I think about it though, the more it makes sense this way, since we can't guarantee that any custom class that extends BaseDataSource would be able to automatically support any new features we add.

Reply:
However, we probably now need to note this function in DEVELOPING.md as one that needs to be implemented. (Can capture that in a follow-up issue if you'd prefer)

Reply:
>However, we probably now need to note this function in DEVELOPING.md as one that needs to be implemented. (Can capture that in a follow-up issue if you'd prefer)

That will be part of https://github.com/elastic/enterprise-search-team/issues/4641